More informative docs for VideoEncoder (#25719)
mdn/content: files/en-us/web/api/videoencoder/reset/index.md

{{APIRef("WebCodecs API")}}{{SecureContext_Header}}

> The **`reset()`** method of the {{domxref("VideoEncoder")}} interface cancels all pending encodes and callbacks, frees all underlying resources and sets the `state` to "unconfigured".

Reviewer: Did you want to say `synchronously` somewhere in here?

After calling `reset()`, `configure()` must be called before resuming `encode()` calls.

> **Note**: To avoid discarding frames queued via `encode()`, `flush()` should be called and completed before calling `reset()`.

Reviewer 2: ```suggestion
> **Note:** To avoid discarding frames queued via `encode()`, `flush()` should be called and completed before calling `reset()`.
```

> > **Note:** To avoid discarding frames queued via {{domxref("VideoEncoder.encode()", "encode()")}}, {{domxref("VideoEncoder.flush()", "flush()")}} should be called and completed before calling {{domxref("VideoEncoder.reset()", "reset()")}}.

Reviewer: "... `flush()` should be used instead of `reset()`"

Author: I assume API users actually want to call reset(), so saying flush should be called instead of it is not helpful

Reviewer: Up to you, this was the point I was trying to make on Tuesday: `flush()` followed by `reset()` isn't something I expect to be that common. It's likely to only happen during mid stream configuration changes.

Author: This scenario seems pretty plausible to me: 
1. get everything out of the encoder
2. reset it to release resources
3. configure later when more encoding work is needed

{{APIRef("WebCodecs API")}}{{SecureContext_Header}}

> Cancells all pending encodes and callbacks, frees all underlying resources and sets the `state` to "unconfigured"

Reviewer 3: add "rejects pending `flush()` promises"

> Cancells all pending encodes and callbacks, frees all underlying resources and sets the `state` to "unconfigured"

Reviewer 2: ```suggestion
The **`reset()`** method of the {{domxref("VideoEncoder")}} interface cancels all pending encodes and callbacks, frees all underlying resources and sets the `state` to "unconfigured".
```

> After calling `reset()`, `configure()` must be called before resuming `encode()` calls.

Reviewer 3: Add {{domxref("VideoEncoder.reset", "reset()")}} for this and other methods?

> **Note**: To avoid discarding frames queued via `encode()`, `flush()` should be called and 
> completed before calling `reset()`.

Reviewer 2: ```suggestion
**Note**: To avoid discarding frames queued via `encode()`, `flush()` should be called and completed before calling `reset()`.
```

## Syntax

> 

Reviewer 3: There is a missing exception in `reset()`. Add:

- `InvalidStateError` {{domxref("DOMException")}}
  - : Thrown if the {{domxref("VideoEncoder.state","state")}} is `"closed"`.

None ({{jsxref("undefined")}}).

### Exceptions

- `InvalidStateError` {{domxref("DOMException")}}

>   - : Thrown if the {{domxref("VideoEncoder.state","state")}} is "closed".

Reviewer 4: ```suggestion
  - : Thrown if the {{domxref("VideoEncoder.state","state")}} is `closed`.
```

Author: I don't think apostrophes ate warranted here. it's a string not a reference to a method or property

Reviewer 4: We are not consistent here, on MDN. Backticks must be added though.
```suggestion
  - : Thrown if the {{domxref("VideoEncoder.state","state")}} is `"closed"`.
```
